src/codegen/codegen_opencl.cc

```diff
@@ -208,6 +208,16 @@ void CodeGenOpenCL::VisitExpr_(const Broadcast* op, std::ostream& os) {   // NOL
   os << "))";
 }
 
+void CodeGenOpenCL::VisitExpr_(const Call *op, std::ostream& os) {  // NOLINT(*)
+  if (op->is_intrinsic(intrinsic::tvm_if_then_else)) {
+    if (op->args[2].type() == UInt(8)) {
```

Comment:
On my environment, the problem also happens with char, short, and ushort.  It looks safe to add a cast for any types.

Reply:
It works with float/int, that's why most people didn't run into this problem. Casting for any types is good for me.